src/py/flwr/server/serverapp/app.py

```diff
@@ -41,38 +65,165 @@ def flwr_serverapp() -> None:
+    )
     args = parser.parse_args()
 
+    certificates = _try_obtain_certificates(args)
+
     log(
         DEBUG,
         "Staring isolated `ServerApp` connected to SuperLink DriverAPI at %s "
         "for run-id %s",
         args.superlink,
         args.run_id,
     )
-    run_serverapp(superlink=args.superlink, run_id=args.run_id)
+    run_serverapp(
+        superlink=args.superlink,
+        run_id=args.run_id,
+        flwr_dir_=args.flwr_dir,
+        certificates=certificates,
+    )
+
 
+def _try_obtain_certificates(
+    args: argparse.Namespace,
+) -> Optional[bytes]:
 
-def run_serverapp(  # pylint: disable=R0914
+    if args.insecure:
+        if args.root_certificates is not None:
+            sys.exit(
+                "Conflicting options: The '--insecure' flag disables HTTPS, "
+                "but '--root-certificates' was also specified. Please remove "
+                "the '--root-certificates' option when running in insecure mode, "
+                "or omit '--insecure' to use HTTPS."
+            )
+        log(
+            WARN,
+            "Option `--insecure` was set. Starting insecure HTTP channel to %s.",
+            args.superlink,
+        )
+        root_certificates = None
+    else:
+        # Load the certificates if provided, or load the system certificates
+        if not isfile(args.root_certificates):
+            sys.exit("Path argument `--root-certificates` does not point to a file.")
+        root_certificates = Path(args.root_certificates).read_bytes()
+        log(
+            DEBUG,
+            "Starting secure HTTPS channel to %s "
+            "with the following certificates: %s.",
+            args.superlink,
+            args.root_certificates,
+        )
+    return root_certificates
+
+
+def run_serverapp(  # pylint: disable=R0914, disable=W0212
     superlink: str,
     run_id: Optional[int] = None,
+    flwr_dir_: Optional[str] = None,
+    certificates: Optional[bytes] = None,
 ) -> None:
-    """Run Flower ServerApp process.
-
-    Parameters
-    ----------
-    superlink : str
-        Address of SuperLink
-    run_id : Optional[int] (default: None)
-        Unique identifier of a Run registered at the LinkState. If not supplied,
-        this function will request a pending run to the LinkState.
-    """
-    _ = GrpcDriver(
+    """Run Flower ServerApp process."""
+    driver = GrpcDriver(
         run_id=run_id if run_id else 0,
         driver_service_address=superlink,
-        root_certificates=None,
+        root_certificates=certificates,
     )
 
-    # Then, GetServerInputs
+    # Resolve directory where FABs are installed
+    flwr_dir = get_flwr_dir(flwr_dir_)
+
+    only_once = run_id is not None
+
+    while True:
+
+        try:
+            # Pull ServerAppInputs from LinkState
+            req = (
+                PullServerAppInputsRequest(run_id=run_id)
+                if run_id
+                else PullServerAppInputsRequest()
+            )
```

Comment:
In protobuf, passing `None` to as argument will be treated as unset.

```suggestion
            req = PullServerAppInputsRequest(run_id=run_id)
```

Reply:
hmm, https://github.com/adap/flower/actions/runs/11562069178/job/32182442604?pr=4370#step:6:33